mdal/mdal.cpp

```diff
@@ -884,15 +884,15 @@ MDAL_DatasetH MDAL_G_addDataset( MDAL_DatasetGroupH group, double time, const do
     return nullptr;
   }
 
-  if ( g->dataLocation() == MDAL_DataLocation::DataOnVolumes )
+  if ( active && g->dataLocation() != MDAL_DataLocation::DataOnVertices )
   {
-    MDAL::Log::error( MDAL_Status::Err_MissingDriverCapability, "Dataset Group has data on 3D volumes" );
+    MDAL::Log::error( MDAL_Status::Err_IncompatibleDataset, "Active flag is only supported on datasets with data on vertices" );
     return nullptr;
   }
 
-  if ( active && g->dataLocation() != MDAL_DataLocation::DataOnVertices )
+  if ( g->dataLocation() == MDAL_DataLocation::DataOnVolumes )
   {
-    MDAL::Log::error( MDAL_Status::Err_IncompatibleDataset, "Active flag is only supported on datasets with data on vertices" );
+    MDAL::Log::error( MDAL_Status::Err_MissingDriverCapability, "Cannot save 3D dataset as a 2D dataset" );
     return nullptr;
   }
```

Comment:
any reason this checks were switched?

Reply:
Laziness in the development process - I deleted the old and create a new one out of order - I will change it back for consistency

Reply:
resolved in bc85fd85d3b09458e2d4ed04ed99644df0138a12